src/main/scala/is/hail/expr/FunctionRegistry.scala

```diff
@@ -1073,9 +1099,59 @@ object FunctionRegistry {
     "nHomVar" -> "Number of samples that are homozygous for the alternate allele."
   )
 
+
+  def chisqTest(c1: Int, c2: Int, c3: Int, c4: Int): (Option[Double], Option[Double]) = {
+
+    var or: Option[Double] = None
+    var chisqp: Option[Double] = None
+
+    if (Array(c1, c2, c3, c4).sum > 0) {
+      if (c1 > 0 && c3 == 0)
+        or = Option(Double.PositiveInfinity)
+      else if (c3 > 0 && c1 == 0)
+        or = Option(Double.NegativeInfinity)
+      else if ((c1 > 0 && c2 > 0 && c3 > 0 && c4 > 0))
+        or = Option((c1 * c4) / (c2 * c3).toDouble)
+        chisqp = Option(chisq.chiSquareTest(Array(Array(c1, c2), Array(c3, c4))))
```

Comment:
Looks like the indentation is wrong.  The assignment to chisqp should just be inside the sum > 0 if, right?